docs/guide/front-end-helpers.md

```diff
@@ -0,0 +1,319 @@
+| `qs` | An object of query string parameters set to values. It does not support recursion. |
+| `body` | The request body. If an object or array it is sent as JSON. Otherwise sent as-is, unless the `send` option is set to `'json'`. |
+| `send` | Set to `'json'` to *always* send the request body as JSON, even if a `FormData` object or non-object. This is not necessary when the body is a normal object. |
+| `parse` | Set to `'json'` to *always* parse the response as JSON. Otherwise the response body is parsed as JSON only if the `Content-Type` is `application/json`. |
+| `headers` | An object containing HTTP header names and values. |
+| `draft` | If `true`, always add `aposMode=draft` to the query string, creating one if needed. |
+| `csrf` | Set to `false` to prevent sending the `X-XSRF-TOKEN` header when talking to the same site. |
+| `fullResponse` | If `true`, return an object with `status`, `headers` and `body` properties, rather than returning the body directly. The individual `headers` are canonicalized to lowercase names. If there are duplicate headers after canonicalizing only the last value is returned. If a header appears multiple times an array is returned for it. |
+| `downloadProgress` | Optional. A function accepting `received` and `total` arguments. It may never be called. If called, `received` will be the bytes sent so far and `total` will be the total bytes to be received. If the total is unknown, it will be `null` |
+| `uploadProgress` | Optional. A function accepting `sent` and `total` arguments. It may never be called. If it is called, `sent` will be the bytes sent so far and `total` will be the total bytes to be sent. If the total is unknown, it will be `null`. |
+
+If the status code is greater than 400 an error is thrown. The error object will be similar to a `fullResponse` object, with a `status` property.
+
+If the URL is site-relative (starts with `/`) it will be requested from the Apostrophe site itself.
+
+::: tip
+Just before the `XMLHttpRequest` is sent, this method emits an event matching the HTTP method. For example, `apos-before-post` for `POST` requests, `apos-before-get` for `GET` requests, etc. The event object has `uri`, `data` and `request` properties. `request` is the `XMLHttpRequest` object.
+
+You can use this to set custom headers on all requests, for example.
+:::
+
+### `parseQuery(query)`
+
+Returns a data object from parsing a URL query string (e.g., `?theme=light&apos-refresh=1`). The argument should not include a URL. The leading question mark (`?`) is allowed but not required. A parameter with no value will be set to `null`.
+
+| Argument | What is it? |
+| -------- | ----------- |
+| `query` | A url query parameter |
+
+```javascript
+const queryParams = apos.http.parseQuery(location.search);
+```
+
+::: note
+`apos.http.parseQuery()` supports query parameter objects and arrays, as well as bracket nesting. Object and array values will be stored as strings and need additional parsing as JSON.
```

Comment:
Objects and arrays will be fully parsed and received as such. However all scalar values (booleans, numbers, etc.) are received as strings, except null which is received as null.